src/axolotl/loaders/model.py

```diff
@@ -214,6 +214,7 @@ def _apply_pre_model_load_setup(self):
         self._set_quantization_config()
         self._set_attention_config()
         self._check_model_requirements()
+        self.patch_manager._apply_patch_deepspeed_zero3()
```

Comment:
You don't need to call it here. Patches from patch manager are called in `self.patch_manager.apply_pre_model_load_patches()` above.

You only need to call your `_apply_patch_deepspeed_zero3` in the `patch_manager.apply_pre_model_load_patches`.